netsim/ansible/templates/evpn/vjunos-switch.j2

```diff
@@ -32,29 +32,48 @@ protocols {
   }
 }
 
-{# VERY DIRTY HACK here for initial testing - since we can specify only one vrf-target per vni-option, let's use the first import one #}
-{# - do it only if vxlan is defined - no need if we are RR only #}
-{% if vxlan is defined %}
-protocols {
-  evpn {
-    encapsulation vxlan;
-    default-gateway no-gateway-community;
-    extended-vni-list all;
-{% if vxlan.vlans is defined %}
-    vni-options {
-{%   for vname in vxlan.vlans if vlans[vname].vni is defined %}
-{%     set vlan = vlans[vname] %}
-        vni {{ vlan.vni }} {
-            vrf-target target:{{ vlan.evpn.import[0] }};
-        }
+{% if vlans is defined and vxlan is defined %}
+vlans {
+{%   for vname,v in vlans.items() if v.evpn.evi is defined and v.evpn.bundle is not defined %}
+  delete: {{ vname }};
+{%   endfor %}
+}
+
+routing-instances {
+{%   for vname,v in vlans.items() if v.evpn.evi is defined and v.evpn.bundle is not defined %}
+  vlan_{{ vname }} {
+    instance-type mac-vrf;
+    service-type vlan-based;
+    vtep-source-interface {{ vxlan.vtep_interface }};
+    route-distinguisher {{ v.evpn.rd }};
+    vrf-target import {% for t in v.evpn.import %}target:{{ t }} {% endfor -%};
```

Comment:
I'm not 100% sure you can have multiple import and export targets. If stuff has not changed, you need to use import and export policies - See https://github.com/ipspace/netlab/blob/dev/netsim/ansible/templates/initial/junos.vrf.j2#L61

Reply:
It's confusing. Multiple targets are accepted, but only one is used :(

Reply:
... added. Now I have to write an integration test to check it out :(